frontend/src/features/rollout-announcement/RolloutAnnouncementModal.tsx

```diff
@@ -0,0 +1,86 @@
+import { useState } from 'react'
+import { BiRightArrowAlt } from 'react-icons/bi'
+import {
+  Box,
+  Modal,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalOverlay,
+  Text,
+} from '@chakra-ui/react'
+
+import Button from '~components/Button'
+
+import { LastFeatureContent } from './components/LastFeatureContent'
+import { NewFeatureContent } from './components/NewFeatureContent'
+import { ProgressIndicator } from './components/ProgressIndicator'
+import { NEW_FEATURES, OTHER_UPDATES } from './Announcements'
+
+interface RolloutAnnouncementModalProps {
+  isOpen: boolean
+  onClose: () => void
+}
+
+export const RolloutAnnouncementModal = ({
+  isOpen,
+  onClose,
+}: RolloutAnnouncementModalProps): JSX.Element => {
+  const [currActiveIdx, setCurrActiveIdx] = useState<number>(0)
+
+  const numOtherUpdatesPages = 1
+  const NUM_NEW_FEATURES = NEW_FEATURES.length + numOtherUpdatesPages
+  const isLastAnnouncement = currActiveIdx === NUM_NEW_FEATURES - 1
```

Comment:
some food for thought: @karrui uses `useMemo` for everything for reasons listed in [this article](https://attardi.org/why-we-memo-all-the-things/), but [some would disagree](https://kentcdodds.com/blog/usememo-and-usecallback). will leave it up to you! (or @karrui if he reviews)

Reply:
Decided not to `useMemo` for this because I think it's not an expensive calculation. But if we want to be consistent as suggested in the [article](https://attardi.org/why-we-memo-all-the-things/), I'll update the PR again